### PR TITLE
[Remove SitesList] Add isSiteCustomizable selector

### PR DESCRIPTION
**Pull request by Tug**
This PR creates the `isSiteCustomizable()` selector and update the `getSite` selector to use it instead of using `getComputedAttribute`. It also updates usages of `isCustomizable()` in components so we can remove `Sites#isCustomizable()` as well as the computed attribute `site.is_customizable`.

### Testing Instructions
- Checkout the branch locally
- Run the tests (`make test`)
- Run the app and check that you don't have access to Themes in the left sidebar when you don't have the permissions.

### CODE REVIEW

File: client/state/sites/selectors.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 	some,
 	split,
 	includes,
-	startsWith,
+	startsWith
```
Comment (dmsnell) [inline client/state/sites/selectors.js]: is this unintentional? seems like an unnecessary change…

File: client/state/sites/selectors.js [hunk 2]
```diff
@@ -63,18 +62,40 @@ export const getSite = createSelector(
...
+	}
+
+	const isWpcomMappedDomain = getSiteOption( state, siteId, 'is_mapped_domain' ) && ! isJetpackSite( state, siteId );
+	const wpcomUrl = withoutHttp( getSiteOption( state, siteId, 'unmapped_url' ) );
+
+	// The 'standard' post format is saved as an option of '0'
+	let defaultPostFormat = getSiteOption( state, siteId, 'default_post_format' );
+	if ( ! defaultPostFormat || defaultPostFormat === '0' ) {
+		defaultPostFormat = 'standard';
+	}
```
Comment (dmsnell) [inline client/state/sites/selectors.js]: tests like this are good reasons for normalizing data form the API on entry into Calypso. instead of scattering around the checks and converting everywhere the data is used it would be helpful if it never came into the system in the odd formats. that's a side-comment :wink:

File: client/state/sites/selectors.js [hunk 3]
```diff
@@ -63,18 +62,40 @@ export const getSite = createSelector(
...
+
+	return {
+		...site.options,
+		...isWpcomMappedDomain && { wpcom_url: wpcomUrl },
+		default_post_format: defaultPostFormat
+	};
+}
+
```
Comment (dmsnell) [inline client/state/sites/selectors.js]: I see that the other new selector is in the `state/selectors` directory but this one is being added to the existing file. do we need to move it?

Comment (Tug) [inline client/state/sites/selectors.js]: Not sure about that. `isSiteCustomizable()` can live on its own now, it's not a computed attribute, whereas `options` still is. We might want to change it later but I think it makes sense to have it alongside `getSite` for now

Comment (dmsnell) [inline client/state/sites/selectors.js]: well we have been adding all new selectors to the selector directory. I should probably rephrase the question: is there a good reason why we _aren't_ adding it to the selector directory instead?

Comment (Tug) [inline client/state/sites/selectors.js]: First, I don't feel strongly on this so if you really think this is a blocker I can simply move it to `state/selectors`.

My logic is that all the selectors used to assemble the site in the `getSite` selector currently lives in that file and I'm not sure we plan to even use this selector at all or if we will always just access `options` through the property of the `site`.
Whereas we keep it as a computed property or not is probably a subject for another PR since I haven't started to update usage of site.options anywhere (and I'm not sure we should).

In the case of `is_customizable`, I was able to update all usage to use the new selector instead and removed it from the site computed properties, so it's definitely a valid/useful selector and this is why it was moved to `state/selectors`.

Comment (dmsnell) [inline client/state/sites/selectors.js]: let's see if @aduth has an opinion on this.

Comment (aduth) [inline client/state/sites/selectors.js]: Having authored the yet-unmerged #10234, I tend to sway more on the extreme of forbidding new selectors outside `state/selectors` directory, regardless of any inconsistencies it might introduce (an expected and necessary pain of adopting a new recommendation).

Comment (mtias) [inline client/state/sites/selectors.js]: Yes, let's put everything new on `state/selectors`.

File: client/state/sites/selectors.js [hunk 4]
```diff
@@ -63,18 +62,40 @@ export const getSite = createSelector(
...
 			domain: getSiteDomain( state, siteId ),
-			is_previewable: isSitePreviewable( state, siteId )
+			is_previewable: isSitePreviewable( state, siteId ),
+			options: getSiteOptions( state, siteId ),
 		};
 	},
 	( state ) => state.sites.items
 );
 
+export function getSiteOptions( state, siteId ) {
```
Comment (mtias) [inline client/state/sites/selectors.js]: Mmm, I actually don't think we should make this an exportable selector. It should only be used in `getSite`. Also probably the name of the function should be changed to reflect we are computing values and not just returning `site.options`.

**Comment by gwwar** [comment]
Changes here look pretty reasonable 👍  Looks like this needs a rebase, ping me when it's ready and I can help smoke test.

**Comment by Tug** [comment]
Rebased and renamed `getSiteOptions ()` into `computeSiteOptions`.

**Comment by gziolo** [comment]
`computeSiteOptions` needs to be promoted to global selector and it's good to go.

Everything else look great 👍

**Comment by Tug** [comment]
I think you don't have access when you have any other role than "Administrator" on a site.
